syntax='proto2';

message Car {
  required string model = 1 [default = "ford"];

  enum BodyType {
    sedan = 0;
    hatchback = 1;
    SUV = 2;
  }

  required BodyType type = 2 [default = sedan];

  optional string color = 3;
  required int32 year = 4 [default = 2000];

  message Owner {
    required string name = 1;
    required string lastName = 2;
    required int64 driverLicense = 3;
  }

  required Owner previousOwner = 5;

  required float a = 6;
}